File: fixtures/ecosystems/python/src/requirements_example.py
"""Implementation stubs for regression test requirements."""

from reqstool_python_decorators.decorators.decorators import Requirements


@Requirements("regression-base-a:REQ_A01")
class RequirementsExample:
    """Implements requirements from all layers."""

    @Requirements("regression-base-b:REQ_B01")
    def measure_perf(self):
        """REQ_B01: Performance measurement."""
        pass

    @Requirements("reqstool-regression:REQ_001")
    def check_reliability(self):
        """REQ_001: Reliability check."""
        pass

    @Requirements("reqstool-regression:REQ_003")
    def validate_security(self):
        """REQ_003: Security validation (deprecated)."""
        pass

    @Requirements("reqstool-regression:REQ_004")
    def maintain_legacy(self):
        """REQ_004: Legacy maintenance (obsolete)."""
        pass

    @Requirements("REQ_L01")
    def interact(self):
        """REQ_L01: Interaction capability."""
        pass

File: fixtures/ecosystems/typescript/src/RequirementsExample.ts
/**
 * @Requirements regression-base-a:REQ_A01
 */
export class RequirementsExample {
  /**
   * @Requirements reqstool-regression:REQ_001
   */
  checkReliability(): boolean {
    return true;
  }

  /**
   * @Requirements regression-base-b:REQ_B01
   */
  measurePerf(): number {
    return 42;
  }

  /**
   * @Requirements reqstool-regression:REQ_003
   */
  validateSecurity(): boolean {
    return true;
  }

  /**
   * @Requirements reqstool-regression:REQ_004
   */
  maintainLegacy(): void {
    // obsolete — no-op
  }

  /**
   * @Requirements REQ_L01
   */
  interact(): string {
    return "interaction";
  }
}
